src/Fhir.Anonymizer.Core/AnonymizationConfigurations/ResourceAnonymizerContext.cs

```diff
@@ -44,18 +44,17 @@ private static void TransformTypeRulesToPathRules(ElementNode node, Dictionary<s
                 return;
             }
 
-            if (typeRules.ContainsKey(node.InstanceType))
+            if (!generatedTypePaths.Contains(path) && typeRules.ContainsKey(node.InstanceType))
```

Comment:
Shall we add some unit tests to explain these fix?

Reply:
Sure. Have added a case to address the problem.